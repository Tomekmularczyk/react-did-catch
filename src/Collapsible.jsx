import React from 'react';
import Types from 'prop-types';

const expandedStyles = {
  overflow: 'hidden',
  transition: 'max-height .3s ease',
  maxHeight: '250px',
};

const collapsedStyles = {
  transition: 'max-height .3s ease',
  overflow: 'hidden',
  maxHeight: '0',
};

const Collapsible = ({ children, isExpanded }) => (
  <div style={isExpanded ? expandedStyles : collapsedStyles}>
    {children}
  </div>
);

Collapsible.propTypes = {
  children: Types.node,
  isExpanded: Types.bool.isRequired,
};

export default Collapsible;