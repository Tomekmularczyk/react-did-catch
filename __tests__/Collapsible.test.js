import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Collapsible from '../src/Collapsible'

describe('<Collapsible />', () => {
  let component
  beforeEach(() => {
    component = shallow(<Collapsible isExpanded={false}>children</Collapsible>)
  })

  it('matches snapshots', () => {
    expect(toJson(component)).toMatchSnapshot()

    component.setProps({ isExpanded: true })
    expect(toJson(component)).toMatchSnapshot()
  })
})
