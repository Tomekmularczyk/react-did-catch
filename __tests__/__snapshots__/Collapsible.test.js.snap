// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Collapsible /> matches snapshots 1`] = `
<div
  style={
    Object {
      "maxHeight": "0",
      "overflow": "hidden",
      "transition": "max-height .3s ease",
    }
  }
>
  children
</div>
`;

exports[`<Collapsible /> matches snapshots 2`] = `
<div
  style={
    Object {
      "maxHeight": "250px",
      "overflow": "hidden",
      "transition": "max-height .3s ease",
    }
  }
>
  children
</div>
`;
